### ISSUE TITLE
UTCDateTime and non-ISO strings

### ISSUE THREAD

**Issue by MMesch** [obspy/obspy#1215]
Hi all,

UTCDateTime misinterprets non-ISO strings without throwing an exception.

Example:

    >> from obspy import UTCDateTime
    >> print UTCDateTime('2015-07-03-06-42')
          2015-07-03T06:04:02.000000Z

An exception if the input string is not valid.

Matthias

**Comment by barsch** [comment]
UTCDateTime supports all kind of non-ISO formats - see https://github.com/obspy/obspy/blob/master/obspy/core/utcdatetime.py#L119 - however if you are working only with ISO date/times you may want to use ```iso8601=True``` during object initialization.

```python
In [1]: from obspy import UTCDateTime

In [2]: print(UTCDateTime('2015-07-03-06-42'))
2015-07-03T06:04:02.000000Z

In [3]: print(UTCDateTime('2015-07-03-06-42', iso8601=True))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-5-f57e3dbd1483> in <module>()
----> 1 print(UTCDateTime('2015-07-03-06-42', iso8601=True))

d:\workspace\obspy\obspy\core\utcdatetime.py in __init__(self, *args, **kwargs)
    248                 if value.count("T") == 1 or iso8601:
    249                     try:
--> 250                         self.timestamp = self._parseISO8601(value).timestamp
    251                         return
    252                     except:

d:\workspace\obspy\obspy\core\utcdatetime.py in _parseISO8601(value)
    413             date_pattern = "%Y%m%d"
    414         else:
--> 415             raise ValueError("Wrong or incomplete ISO8601:2004 date format")
    416         # check for time zone information
    417         # note that the zone designator is the actual offset from UTC and

ValueError: Wrong or incomplete ISO8601:2004 date format
```

**Comment by MMesch** [comment]
ok thanks that's good to know! But don't you think that:  non-iso 2015-07-03-06-42 should be interpreted as 2015-07-03T06:42:00.000000Z rather than 2015-07-03T06:04:02.000000Z , which should probably be 2015-07-03-06-04-02. That's maybe why standards exist :)

**Comment by megies** [comment]
I think we should probably either raise an Exception for this or try harder to parse it correctly.. in any case, how this is parsed currently is pretty strange.

**Comment by barsch** [comment]
this boils down to:
```python
(Pdb) import datetime
(Pdb) datetime.datetime.strptime(u'201507030642', u'%Y%m%d%H%M%S')
datetime.datetime(2015, 7, 3, 6, 4, 2)
(Pdb) datetime.datetime.strptime(u'20150703064200', u'%Y%m%d%H%M%S')
datetime.datetime(2015, 7, 3, 6, 42)
```
so zero padding as documented in https://docs.python.org/2/library/datetime.html#strftime-strptime-behavior is not enforced in strptime :/ So I guess we have to introduce different paddings depending on the string size in https://github.com/obspy/obspy/blob/master/obspy/core/utcdatetime.py#L284-L287

**Comment by krischer** [comment]
Fixed in #1221
